src/main/java/com/lru/LRU.java

```diff
@@ -79,20 +79,17 @@ String display(){
     }
 
     private Node<K,V> insertIntoHead(Node<K,V> node) {
-        try {
-            lock.lock();
-            Node curNext = head.next;
-            head.next = node;
-            node.prev = head;
-            node.next = curNext;
-            curNext.prev = node;
-            return node;
-        } finally {
-            lock.unlock();
-        }
-
+        Node curNext = head.next;
+        head.next = node;
+        node.prev = head;
+        node.next = curNext;
+        curNext.prev = node;
+        return node;
     }
 
+    /**
+     * keeping the locks on this private method - as this could arguably made public
+     */
     private void cleanUp() {
         try {
             lock.lock();
```

Comment:
You do not need a while loop as, if the capacity goes up by one it will do the cleanup and while loop will always run just once